From 12eadfb7b821711c45d24e16b6c082ab904f7bfd Mon Sep 17 00:00:00 2001
From: Johan Hovold <johan+linaro@kernel.org>
Date: Sun, 31 Jul 2022 13:43:46 +0200
Subject: [PATCH 155/230] hack: arm64: dts: qcom: sc8280xp-x13s: enable both
 touchpads

Enable both touchpad nodes in the devictree and let the HID driver
determine which one is actually populated.

Note that the pin configuration must currently be moved to the parent
i2c-bus node even though only one of these nodes will ever be
successfully probed on a specific device.

This is needed until the boot firmware can enable only the node for the
populated touchpad.

Not-signed-off-by: Johan Hovold <johan+linaro@kernel.org>
---
 .../boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts    | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
index 372814707c0..a4897efbf59 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
@@ -836,7 +836,7 @@ &qup2_i2c5 {
 	clock-frequency = <400000>;
 
 	pinctrl-names = "default";
-	pinctrl-0 = <&qup2_i2c5_default>;
+	pinctrl-0 = <&qup2_i2c5_default>, <&tpad_default>;
 
 	status = "okay";
 
@@ -848,12 +848,7 @@ touchpad@15 {
 		interrupts-extended = <&tlmm 182 IRQ_TYPE_LEVEL_LOW>;
 		vdd-supply = <&vreg_misc_3p3>;
 
-		pinctrl-names = "default";
-		pinctrl-0 = <&tpad_default>;
-
 		wakeup-source;
-
-		status = "disabled";
 	};
 
 	touchpad@2c {
@@ -864,9 +859,6 @@ touchpad@2c {
 		interrupts-extended = <&tlmm 182 IRQ_TYPE_LEVEL_LOW>;
 		vdd-supply = <&vreg_misc_3p3>;
 
-		pinctrl-names = "default";
-		pinctrl-0 = <&tpad_default>;
-
 		wakeup-source;
 	};
 
-- 
2.39.0

